File: designPattern/template/python/AbstractPrint.py
import abc


class AbstractPrint:
    __metaclass__ = abc.ABCMeta

    @abc.abstractmethod
    def open(self):
        pass

    @abc.abstractmethod
    def print_input(self):
        pass

    @abc.abstractmethod
    def close(self):
        pass

    def display(self):
        self.open()
        for i in range(0, 5):
            self.print_input()
        self.close()

File: designPattern/template/javascript/AbstractPrint.js
export default class AbstractPrint {
  open() {
    throw new Error("this is AbstractMethod");
  }

  print() {
    throw new Error("this is AbstractMethod");
  }

  close() {
    throw new Error("this is AbstractMethod");
  }

  display() {
    open();
    for (const i in [...Array(5).keys()]) {
      print();
    }
    close();
  }
}
